src/api/records.test.tsx

```diff
@@ -0,0 +1,93 @@
+import { Record, RecordRow } from '../app.types';
+import {
+  testRecords,
+  testRecordRows,
+  renderWithProvidersForHook,
+} from '../setupTests';
+import axios from 'axios';
+import { renderHook, waitFor } from '@testing-library/react';
+import { useRecordCount, useRecordsPaginated } from './records';
+
+const dataResponsesEqual = (x?: RecordRow[], y?: RecordRow[]): boolean => {
+  if (!x || !y) return false;
+  if (x.length !== y.length) return false;
+
+  for (let i = 0; i < x.length; i++) {
+    const xRow = x[i];
+    const yRow = y[i];
+
+    const xKeys = Object.keys(xRow);
+    const yKeys = Object.keys(yRow);
+
+    for (let i = 0; i < xKeys.length; i++) {
+      if (xKeys[i] !== yKeys[i]) return false;
+    }
+
+    if (xRow.timestamp !== yRow.timestamp) return false;
+    if (xRow.shotnum !== yRow.shotnum) return false;
+    if (xRow.activeArea !== yRow.activeArea) return false;
+    if (xRow.activeExperiment !== yRow.activeExperiment) return false;
+  }
+
+  return true;
+};
+
+describe('records api functions', () => {
+  let mockData: Record[];
+
+  beforeEach(() => {
+    mockData = testRecords;
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('useRecordsPaginated', () => {
+    it('sends axios request to fetch records and returns successful response', async () => {
+      (axios.get as jest.Mock).mockResolvedValue({
+        data: mockData,
+      });
+      // params.append('limit', '25');
+      // params.append('skip', '0');
+
+      const { result } = renderHook(() => useRecordsPaginated(), {
+        wrapper: renderWithProvidersForHook(),
```

Comment:
@louise-davies I tried to use your `renderWithProviders` function here but couldn't get the syntax right. What parameters should I pass to it if I don't care about rendering any child elements to the screen?

Reply:
Ah, I'm not sure if it works without a child component. It's not really designed for being used as a wrapper for hooks - I pretty much stole it from the redux docs 😅 

Looking at the DG tests, I think `wrapper` here expects just the wrapping component, and not a `render`. We could probably extract out the wrapper itself to export for hooks testing, and then have another function with the render for use for component testing.

Reply:
That sounds good to me, I can modify what I've done here

Reply:
Had a go at fixing it but clearly the end of the week has come and my brain hasn't hacked it. So I've stuck with renaming the render functions to improve their clarity and we can always come back to it later